subprojects/docs/src/docs/userguide/java.adoc

```diff
@@ -0,0 +1,453 @@
+
+Specifying the dependencies for your Java project requires just three pieces of information:
+
+ * Which dependency you need, such as a name and version
+ * What it's needed for, e.g. compilation or running
+ * Where to look for it
+
+The first two are specified in a `dependencies {}` block and the second in a `repositories {}` block. For example, to tell Gradle that your project requires version 3.6.7 of http://hibernate.org/[Hibernate] Core to compile and run your production code, and that you want to download the library from the Maven Central repository, you can use the following fragment:
+
+++++
+<sample id="javaPluginDependencyMgmt" dir="userguide/java/basic" title="Declaring dependencies">
+    <sourcefile file="build.gradle" snippet="java-dependency-mgmt"/>
+</sample>
+++++
+
+The Gradle terminology for the three elements is as follows:
+
+ * _Repository_ (ex: `mavenCentral()`) — where to look for the modules you declare as dependencies
+ * _Configuration_ (ex: `implementation`) - a named collection of dependencies, grouped together for a specific goal such as compiling or running a module — a more flexible form of Maven scopes
+ * _Dependency coordinate_ (ex: `org.hibernate:hibernate-core-3.6.7.Final`) — the ID of the dependency, usually in the form '__<group>__:__<module>__:__<version>__' (or '__<groupId>__:__<artifactId>__:__<version>__' in Maven terminology)
+
+You can find a more comprehensive glossary of dependency management terms <<dependency_management_terminology,here>>.
+
+As far as configurations go, the main ones of interest are:
+
+ * `compileOnly` — for dependencies that are necessary to compile your production code but shouldn't be part of the runtime classpath
+ * `implementation` (supersedes `compile`) — used for compilation and runtime
+ * `runtimeOnly` (supersedes `runtime`) — only used at runtime, not for compilation
+ * `testCompileOnly` — same as `compileOnly` except it's for the tests
+ * `testImplementation` — test equivalent of `implementation`
+ * `testRuntimeOnly` — test equivalent of `runtimeOnly`
+
+You can learn more about these and how they relate to one another in the <<sec:java_plugin_and_dependency_management,plugin reference chapter>>.
+
+Be aware that the <<java_library_plugin,Java Library Plugin>> creates an additional configuration — `api` — for dependencies that are required for compiling both the module and any modules that depend on it.
+
+[NOTE]
+.Why no `compile` configuration?
+====
+The Java Plugin has historically used the `compile` configuration for dependencies that are required to both compile and run a project's production code. It is now deprecated because it doesn't distinguish between dependencies that impact the public API of a Java library project and those that don't. You can learn more about the importance of this distinction in _<<sec:building_java_libraries,Building Java libraries>>_.
```

Comment:
I was not aware that `compile` is deprecated. My understanding is that it's discouraged, but needed for certain situations. Maybe worth a double-check, but otherwise I trust you've gotten an authoritative answer from others on this.